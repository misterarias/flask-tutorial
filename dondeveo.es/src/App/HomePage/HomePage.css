body {
  font-family: Roboto, helvetica, arial, sans-serif;
}

.border {
  height: 100vh;
  border: 0 ;
}
