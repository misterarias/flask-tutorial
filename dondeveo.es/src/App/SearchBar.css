.container {
  text-align: center;
  background-color: white;
  padding: 10px;
  margin: 60px;
}

.header {
  height: 60px;
  padding: 10px;
  color: #222;
}

.title {
  font-size: 1.5em;
}

.searchbar input {
  width: 150px;
  height: 40px;
  padding-left: 5px;
  margin: 0 5px;
  font-size: 16px;
}
.searchbar button {
  height: 40px;
  font-size: 16px;
}
