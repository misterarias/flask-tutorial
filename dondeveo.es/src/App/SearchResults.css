.search-results {
  background: #cebada;
}

.img-circle {
  /* make it responsive */
  max-width: 100%;
  width: 100%;
  /* height: 100%; */
  display: block;

  /* make it a circle */
  padding: 50%;
  background: #e1e1e1;
  border-radius: 50%;

  /* Centering on image`s center*/
  background-position-y: center;
  background-position-x: center;
  background-repeat: no-repeat;

  /* it makes the clue thing, takes smaller dimension to fill div */
  background-size: cover;

  /* it is optional, for making this div centered in parent*/
  margin: 0 auto;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
}
