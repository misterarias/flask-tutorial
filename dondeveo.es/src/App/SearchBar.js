import React, { Component } from 'react';
import { Input, Button, Container, Col, Row, Jumbotron } from 'reactstrap';

import './SearchBar.css';

export default class SearchBar extends Component {

  constructor() {
    super();

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);

    this.state = {
      query: ""
    }
  }

  handleChange(ev) {
    this.setState({
      query: ev.target.value
    });
  }

  handleSubmit(ev) {
    this.props.onSubmit(this.state.query);
  }

  render() {
    return (
      <div>
        <Jumbotron>
          <Container>
            <h1 className="display-3">¿Dónde veo.... ?</h1>

            <p className="lead">
              Busca aquí dónde ver la peli que te apetezca
            </p>

          </Container>
        </Jumbotron>

        <Row >
          <Col sm={10}>
            <Input
              type="text"
              name="searchTerm"
              onChange={this.handleChange}
              placeholder="Título de película..."
            />
          </Col>

          <Col sm={2}>
            <Button
              block
              color="primary"
              onClick={this.handleSubmit}
            > Search </Button>
          </Col>
        </Row>
      </div>
    )
  }
}
