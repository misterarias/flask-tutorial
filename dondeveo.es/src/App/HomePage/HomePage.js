import React, { Component } from 'react';

import SearchBar from '../SearchBar/SearchBar' ;
import SearchResults from '../SearchResults/SearchResults' ;
import Search from '../Api/Search' ;

export default class HomePage extends Component  {

  constructor() {
    super();

    this.state = {
      results: []
    }
  }

  doSearch(searchTerm) {
    const movieSet =  Search.results();
    if (movieSet.total_results === 0) return []
    return movieSet.results.map( (item, key) => {
      return {
        'name': item.original_title,
        'description': item.overview,
        'thumbnail': item.poster_path,
        'year': item.release_date
      }
    });
  }

  handleSubmit(searchTerm) {
    const searchResults = this.doSearch(searchTerm);
    this.setState({
      results: searchResults
    });
  }

  render() {
    return <div>
      <SearchBar
        handleSubmit={(searchTerm) => this.handleSubmit(searchTerm)} />

      <SearchResults
        results={this.state.results}
      />
    </div>
  }
}

