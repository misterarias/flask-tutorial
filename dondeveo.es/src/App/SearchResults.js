import React, { Component } from 'react';
import './SearchResults.css';

class Movie extends Component {
  render() {
  return(
    <div className="result" onClick={() => this.props.handleClick(this.props.name)}>
      <h3>{this.props.name}</h3>
      <p>{this.props.description}</p>
    </div>
  );
  }
}

export default class SearchResults extends Component {

  constructor(props) {
    super();

    this.state = {
      results: props.results || []
    };
  }

  render() {
    const results = this.state.results.map( (item, key) => {
      return (
        <li key={key}>
          <Movie
            name={item}
            handleClick={this.props.handleClick}
            description="Created in a map()"
          />
        </li>
      );
    });

    return (
      <div className="search-results">
        <ul id="results">{results}</ul>
      </div>
    );
  }
}
