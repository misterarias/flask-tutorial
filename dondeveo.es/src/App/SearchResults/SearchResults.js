import React, { Component } from 'react';
import { Media, Row, Col, ListGroup, ListGroupItem, ListGroupItemHeading, ListGroupItemText } from 'reactstrap';
import ApiInfo from '../Api/ApiUtils' ;
import './SearchResults.css';

import { Route } from 'react-router-dom' ;

class Movie extends Component {
  render() {
    const movie = this.props.movie;

    let year = this.props.movie.year ;
    if (year.length > 4) {
      year = year.substr(0, 4);
    }

    let thumbnail = movie.thumbnail;
    if (thumbnail) {
      thumbnail = new ApiInfo().getImagePath(thumbnail);
    }

    return <Route render={ ({history}) => (
      <ListGroupItem
        action={true}
        className="listItem"
        onClick={ () => { history.push(`/movie/${movie.name}`); } }
      >
        <Row>
          <Col sm={2}>
            <Media object src={thumbnail} className="img-circle" />
          </Col>
          <Col sm={10}>
            <ListGroupItemHeading>{movie.name} <i className="small">({year})</i></ListGroupItemHeading>
            <ListGroupItemText>{movie.description}</ListGroupItemText>
          </Col>
        </Row>

      </ListGroupItem>
    )
    } />
  }
}

class SearchResults extends Component {


  render() {
    const results = this.props.results || [] ;
    const movieList = results.map( (item, key) => {
      return (
        <Movie
          key={key}
          movie={item}
        />
      );
    });

    return (
      <Row>
        <Col sm={12}>

          <p className="lead">
            { results.length > 0 ?`Mostrando ${results.length} resultados` : '' }
          </p>

          <ListGroup className="search-results">
            {movieList}
          </ListGroup>

        </Col>
      </Row>
    );
  }
}

export default SearchResults;
