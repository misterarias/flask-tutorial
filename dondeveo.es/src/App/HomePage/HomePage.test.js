import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from './HomePage';
import SearchResults from '../SearchResults/SearchResults';
import SearchBar from '../SearchBar/SearchBar' ;

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() })

it('renders without crashing', () => {
  const tree = renderer.create(
    <HomePage />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

test('HomePage has the expected elements', () => {
  const app = shallow(
    <HomePage />
  );

  expect(app.find(SearchBar).length).toBe(1);
  expect(app.find(SearchResults).length).toBe(1);
});
