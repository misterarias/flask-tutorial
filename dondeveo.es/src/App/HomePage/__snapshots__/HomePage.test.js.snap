// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    className="row"
    style={
      Object {
        "marginBottom": "1rem",
      }
    }
  >
    <div
      className="col col-sm-10"
    >
      <input
        className="form-control"
        name="searchTerm"
        onChange={[Function]}
        placeholder="Título de película..."
        type="text"
      />
    </div>
    <div
      className="col col-sm-2"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
         Search 
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col col-sm-12"
    >
      <p
        className="lead"
      >
        
      </p>
      <ul
        className="search-results list-group"
      />
    </div>
  </div>
</div>
`;
